programs/psa/psa_hash.c

```diff
@@ -33,18 +33,24 @@
 #include "mbedtls/build_info.h"
 #include "mbedtls/platform.h"
 
+/* The algorithm used by this demo is SHA 256.
+ * Please see include/psa/crypto_values.h to see the other
```

Comment:
We should probably link to the part of the PSA Crypto API spec relating to hashes (I guess that's Section 10.2). It's quite terse reference information, but should be mentioned.

However, since we're writing an example program, just linking to the spec is possibly going to have someone set `HASH_ALG` to `PSA_ALG_SM3` and be annoyed that it doesn't even compile.

So why not link to both?

Start with a link to the chapter in the spec on hash algorithms, and then say something along the lines of `see include/psa/crypto_values.h for the algorithm values known to Mbed TLS`.

And thinking about that, the code should also clearly output when an unsupported algorithm is given.